### PR TITLE
Fix rounded border overflow in Safari

### LINKED ISSUE #36395: SPT: Template thumbnail border-radii looks strange
image

Seems to be some sort of overflow issue, or mis-matched border-radii. Seems to be only in the latest Safari.

### PR DESCRIPTION
**Pull request by WunderBart**
#### Changes proposed in this Pull Request

For the starter page templates (thumbnails) previews use the `translate3d(0, 0, 0)` trick to help Safari render rounded borders better.

#### Testing instructions

- Check out this branch
- Make sure Full Site Editing & Gutenberg plugins are active
- Open Safari & go to `/wp-admin/post-new.php?post_type=page`
- Look at the template preview thumbnail borders - they shouldn't have overflown or thinner corners.

**Before:**

image

**After:**

Screenshot 2019-11-04 14 51 37

Fixes #36395

### CODE REVIEW

**Comment by shaunandrews** [comment]
Thank you! 🚢

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss [hunk 1]
```diff
@@ -169,6 +169,7 @@ body.is-fullscreen-mode {
 		overflow: hidden;
 		background-color: #fff;
 		position: relative;
+		transform: translate3d( 0, 0, 0 ); // Fix for Safari rounded border overflow (1/2).
```
Comment (getdave) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: Love to see the comments for extra context 👍 

I am interested in _why_ this fixes it. Why does it happen and why is this necessary to fix it.

Comment (WunderBart) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: > (...) why this fixes it. Why does it happen (...)

Great questions. It seems to be related to how Safari processes [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) (which is different than Chrome's). I'm lacking a deep understanding of how exactly that process is applied, so maybe someone could help out here. I bumped into similar issues with Safari before and, unless the explicit `z-index` did not work (which is the case here), the `translate3d(0, 0, 0)` did the trick.  It applies another level to the stacking context that acts as an encapsulator, separating the respective stacking contexts of the involved elements ([source](https://medium.com/@mattcroak718/understanding-the-stacking-context-for-cross-browser-compatibility-2b21db1cf621)). 

I've just realized that the `translateZ(0)` should work as well and would fit better here. Will update.

> (...) why is this necessary to fix it.

This looks like a _hack_, but Safari is known to have issues like this one. We probably could try rebuilding the whole component to try different techniques that wouldn't require forced `transform` to make it look good in Safari, but I'm not sure if it's worth the effort in this case. Does that make sense?

Comment (WunderBart) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: For the larger context, I was also seeing (before the fix) other random artifacts, like this one:

**Comment by marekhrabe** [comment]
👏

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss (hunk 1 above)
Comment (getdave) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/styles/starter-page-templates-editor.scss]: TIL. All good by me. Thanks for the detailed response.
